refarch/aws-native/cicd/pipeline.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+from aws_cdk.core import Stack, Construct, Environment
+from aws_cdk.pipelines import CodePipeline, CodePipelineSource, CodeBuildStep
+
+from cicd.pipeline_stage import PipelineStage
+
+
+class AnalyticsEnvironment:
+
+    def __init__(self, name: str, account: str, region: str):
+        self.name = name
+        self.account = account
+        self.region = region
+
+
+class PipelineStack(Stack):
+
+    def __init__(self, scope: Construct, id: str, deploy_envs: list, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        connection_arn = self.node.try_get_context('ConnectionArn')
+        repo = self.node.try_get_context("RepositoryName")
+        branch = self.node.try_get_context("RepositoryBranch")
+
+        if connection_arn is None or connection_arn == '<CONNECTION_ARN>':
```

Comment:
Shouldn't we create the repo automatically if not provided : If not given then create code commit repo ? (As they do in retail store https://github.com/aws-samples/retail-demo-store)

Reply:
@flochaz  As discussed during the call, let's check with Vincent.


@vgkowski  What's your thought on this? Two options: 
1. Default to creating CodeCommit for users if CONNECTION_ARN isn't provided.
2. Throw an error as it is. Only support Connection as a repo source.

The trade-off here is around code complexity and user experience.  If you don't expect customers to use CodeCommit (and use mostly Github fork), I suggest 2. But if we expect them to fork this to their CodeCommit repo and develop further. Option 1 is preferred. 

For option 1 implementation, we'll copy the code from local machine to S3 (with [s3-deployment](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-s3-deployment-readme.html) and use a custom resource to initalize the CodeCommit repo.  Effort ~= 2-4 days.